### DISCUSSION TITLE
How do I develop locally?

### DISCUSSION THREAD

**Discussion by designbyadrian** [google/model-viewer#3944] [Q&A]
Hello!

In `packages/model-viewer`, there's a Node script `dev` that runs typescript and rollup in watch mode.

How do I preview the changes I make to model-viewer as watch mode is running?

In the root, there's `build` and `serve`. I can view the model-viewer.dev, but nothing happens if I modify any code. The references to model-viewer are only for the node-modules package, and if I change it to the local `dist` folder of model-viewer, I have to stop the server, run build again, and serve again to see any changes.

**Comment by balintkissshapr3d** [comment]
You can run bootstrap -> build -> serve from the root, open modelviewer.dev as you mentioned and after you modify the code you can run build again (no need the restart the server)

**Reply by designbyadrian** [reply]
Thanks, should have guessed that one! There's a lot of building going on though, nowhere to cut corners, and no watch mode?

**Reply by elalish** [reply]
try `npm run build:dev` to shorten the build process. You might also be interested in `npm run dev`. Look though the packages/model-viewer/package.json scripts.
